clash-cores/src/Clash/Cores/LineCoding8b10b.hs

```diff
@@ -88,22 +88,16 @@ encode8b10b ::
   (Bool, BitVector 10)
 encode8b10b rd dw = out
  where
-  cw = case dw of
-    Cw _ -> 1
-    _ -> 0
+  cw = isValidSymbol dw && not (isDw dw)
 
   (statusBits, cg) =
     splitAt d2
       $ bv2v
       $ asyncRomBlobPow2
         $(memBlobTH Nothing Enc.encoderLut)
-      $ unpack (cw ++# bitCoerce rd ++# fromDw dw)
+      $ unpack (bitCoerce cw ++# bitCoerce rd ++# fromDw dw)
```

Comment:
```suggestion
      $ unpack (pack cw ++# pack rd ++# pack dw)
```
This is more idiomatic. Another trick would be:
```
$ bitCoerce (cw, rd, dw)
```
using the fact that `++#` and tuple elements have an identical effect on what the `BitVector` looks like.